code/modules/projectiles/guns/energy.dm

```diff
@@ -33,9 +36,29 @@
 	shot = ammo_type[select]
 	fire_sound = shot.fire_sound
 	fire_delay = shot.delay
+	SSobj.processing |= src

```

Comment:
This adds it to proccessing objects even if it's not self charging, which means all you have to do is edit selfcharge to 1 to get it to recharge.